options.go

```diff
@@ -0,0 +1,66 @@
+package tflog
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type Option func(loggerOpts) loggerOpts
+
+type loggerOpts struct {
+	name            string
+	level           hclog.Level
+	includeLocation bool
+
+	// some private options to be used only by tflog for testing purposes
+	// we should never export an Option that sets these
+	output io.Writer
+}
+
+func applyLoggerOpts(opts ...Option) loggerOpts {
+	// set some defaults
+	l := loggerOpts{
+		level:           hclog.Trace,
+		includeLocation: true,
+	}
+	for _, opt := range opts {
+		l = opt(l)
+	}
+	return l
+}
+
+func withOutput(output io.Writer) Option {
+	return func(l loggerOpts) loggerOpts {
+		l.output = output
+		return l
+	}
+}
+
+func WithLogName(name string) Option {
+	return func(l loggerOpts) loggerOpts {
+		l.name = name
+		return l
+	}
+}
+
+func WithLevelFromEnv(name string, subsystems ...string) Option {
+	return func(l loggerOpts) loggerOpts {
+		envVar := strings.Join(subsystems, "_")
+		if envVar != "" {
+			envVar = "_" + envVar
+		}
+		envVar = strings.ToUpper("TF_LOG_PROVIDER_" + name + envVar)
```

Comment:
Is this function expected to be used outside Providers? If not, should it include Provider in the naming? If so, should `_PROVIDER` be handled separately?

Reply:
It's probably scoped to "plugins", which I suppose isn't "providers". I wanted to add a convenience function here to enforce some consistency, but I suppose we can do that in the SDK code that instantiates and injects the logger.